src/components/authorization/authorization.service.ts

```diff
@@ -104,6 +109,37 @@ export class AuthorizationService {
     return parseSecuredProperties(props, permissions, resource.SecuredProps);
   }
 
+  async verifyCanEditChanges<
+    TObject extends Resource,
+    Key extends keyof TObject & string
+  >(
+    baseNode: TObject,
+    props: readonly Key[],
+    changes: { [Key in keyof TObject]?: UnwrapSecured<TObject[Key]> }
+  ) {
+    for (const prop of props) {
+      if (
+        changes[prop] === undefined ||
+        unwrapSecured(baseNode[prop]) === changes[prop]
```

Comment:
This needs to be smarter for DateTime's and string lists which will fail identity checks (`===`) even when the value the represent is the same.